### PR TITLE
LGAT subtyping fix Part3: Re run lgat subtyping

### PR DESCRIPTION
**Pull request by kgaonkar6**
#### What scientific question is your analysis addressing?
Re-run of LGAT subtyping with edits from #978 #979 and in the process realized that edits to analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd were required without which it would error out. 

#### What was your approach?
Edits to analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd  were to load GenomicRanges specifically and also a line was missing  which is now added:
https://github.com/kgaonkar6/OpenPBTA-analysis/blob/8dc5a01c62a2675a95ad699228bac70d8d696638/analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd#L245

#### What GitHub issue does your pull request address?
#970 #971 

#### Which areas should receive a particularly close look?
I think I should update the  bash script to run all scripts through CI to capture any errors , I had it not running in CI before which would have captured the issue with analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd. 

#### What types of results are included (e.g., table, figure)?
table

#### What is your summary of the results?
From the MYB fusion and IDH mutation updates in the previous PRs, the subtype in analyses/molecular-subtyping-LGAT/results/lgat_subtyping.tsv have been updated.

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
> I think I should update the bash script to run all scripts through CI to capture any errors , I had it not running in CI before which would have captured the issue with analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd.

Apologies if we've had this conversation before - is the issue that there are not enough LGAT samples in the CI subset for the first 3 steps to run without error and that's why things are set up as is?

**Comment by kgaonkar6** [comment]
Hmm so I didn't quite remember the exact reason to why I have the subset scripts only running locally and not in CI but I can track to [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/642/files#r396117128) and I kept the same logic  when we revamped the sub-typing to be more inclusive of other non-BRAF subtypes.

**Comment by jaclyn-taroni** [comment]
I guess our question boils down to: What happens if you don't skip those steps in CI?

**Comment by kgaonkar6** [comment]
right, I can remove the logic and add the script here to find out.

**Comment by jaclyn-taroni** [comment]
Let's see what happens 🎲

**Comment by kgaonkar6** [comment]
Hmm so the odds are not in our favor .. it failed looking for BRAF V600E mutation in the first script . Since we have so many LGATs I think the issue must be with the consensus maf file in testing not having any BRAF V600Es ? I can look into this ..

**Comment by kgaonkar6** [comment]
These are the only snvs that were gathered in the 1st first as subtype defining mutations which is why the code errors out.
Tumor_Sample_Barcode<chr> | IMPACT<chr> | SIFT<chr> | PolyPhen<chr> | Hugo_Symbol<chr> | Variant_Classification<chr>
-- | -- | -- | -- | -- | --
BS_7M7JNG00 | MODIFIER | . | . | MET | Intron
BS_7M7JNG00 | HIGH | . | . | NF1 | Nonsense_Mutation

**Comment by jaclyn-taroni** [comment]
Before we take next steps, we should find out if the other steps also fail. I'd put the SNV step back in the logic for now.

**Comment by kgaonkar6** [comment]
Fusion subset works fine, cnv subsetting script will error out in CI because manta in testing data doesn't have any FGFR1 SVs. So should I keep the all subsetting scripts as they were within the CI logic?

**Comment by jaclyn-taroni** [comment]
It seems like, from your description, having this module run in CI would have caught some things. An alternative to including that logic would be enrich for specific lesions (like we do for fusions: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/5e18a406999fb40abf5ed46ee319a2b117864acb/analyses/create-subset-files/02-subset_files.R#L112) or make sure we include specific samples (like we do for the NF1 and TP53 classifier: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/5e18a406999fb40abf5ed46ee319a2b117864acb/analyses/create-subset-files/01-get_biospecimen_identifiers.R#L192).

**Comment by jaclyn-taroni** [comment]
I filed #985 to keep track of changing the files in CI, should we want to go that route. For now, I think I'd put the logic back in and resolve the conflicts and I will go ahead and review this!

**Comment by kgaonkar6** [comment]
while going through 01-subset-files-for-LGAT.Rmd  to resolve the conflict I also noticed that some quick fixes to the code here : 
https://github.com/kgaonkar6/OpenPBTA-analysis/blob/f122cd9cce40fa189e1ae902f759b85f8f18b104/analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd#L271

could help pass 01-subset-files-for-LGAT.Rmd through CI
```
dplyr::mutate(BRAF_V600E_mut = if_else(rowSums(
                  dplyr::select(.,snvOI$BRAF_V600E$gene)) >=1,"Yes","No")

```
since I actually have similar conditions for other genes in the code or would it be better to update the `create-subset-files` instead?

With the above updated 01 and 02 scripts will pass through CI, 03 will need a little more updates (for example only merge `overlaps_manta_fgfr_df` if it is a non-empty df) if we want to go through this path.

**Comment by jaclyn-taroni** [comment]
I need a little more context before weighing in @kgaonkar6. Is there a reason we didn't use `rowSums()` before? In general, I'd prefer to make the code more robust over changing the CI files.

**Comment by kgaonkar6** [comment]
I added the check for column name and rowSums() logic for genes that I knew didn't have mutations in our filtered `consensusMutationSubset` like HIST1H3C, HIST2H3C, or H3F3A mutations here : 
https://github.com/kgaonkar6/OpenPBTA-analysis/blob/f122cd9cce40fa189e1ae902f759b85f8f18b104/analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd#L279-L284

But didn't add it to all other genes like BRAF, FGFR1 etc which have mutation calls so could be accessed directly by checking for presence.

I believe adding the check for gene in column names plus rowSums()  to all genes will make the code more robust and pass CI.

**Comment by jaclyn-taroni** [comment]
> I believe adding the check for gene in column names plus rowSums() to all genes will make the code more robust and pass CI.

Okay, great, let's give this a try!

File: analyses/molecular-subtyping-LGAT/run_subtyping.sh [hunk 1]
```diff
@@ -9,10 +9,17 @@ cd "$(dirname "${BASH_SOURCE[0]}")"
...
-  Rscript -e "rmarkdown::render('03-subset-cnv-files-LGAT.Rmd')"
+# subset bu SNV 
+Rscript -e "rmarkdown::render('01-subset-files-for-LGAT.Rmd')"
+# subset by Fusion
+Rscript -e "rmarkdown::render('02-subset-fusion-files-LGAT.Rmd')"
+
 
+if [ "$SUBSET" -gt "0" ]; then
+# subset by CNV
+Rscript -e "rmarkdown::render('03-subset-cnv-files-LGAT.Rmd')"
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/run_subtyping.sh:20]:
```suggestion
  Rscript -e "rmarkdown::render('03-subset-cnv-files-LGAT.Rmd')"
```

**Review by jaclyn-taroni — APPROVED** [review]
This looks good to me. I had one suggestion about how to shorten the changes to the call to `mutate()` with a custom function, but I don't feel strongly that that change needs to be made.

File: analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd [hunk 2]
```diff
@@ -267,27 +267,35 @@ H3.3_mut | harbors an H3F3A K28M or G35R/V mutation
...
-  dplyr::mutate(BRAF_V600E_mut = if_else(BRAF>=1,"Yes","No"),
-                FGFR_mut = if_else(rowSums(
-                  dplyr::select(.,snvOI$FGFR$gene)) >=1,"Yes","No"),
-                IDH_mut  = if_else(any(colnames(.) %in% snvOI$IDH$gene) & 
-                                     rowSums(dplyr::select(.,one_of(snvOI$IDH$gene))) >=1,
+  reshape2::dcast(Kids_First_Biospecimen_ID + sample_id~ Hugo_Symbol,
+                  fun.aggregate = length) %>%
+  dplyr::mutate(BRAF_V600E_mut = if_else(any(colnames(.) %in% snvOI$BRAF_V600E$gene) & 
+                                     rowSums(dplyr::select(.,one_of(snvOI$BRAF_V600E$gene))) >=1,
+                                     "Yes","No"),
```
[the remark this answers is not in the document]
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd]: That was a beautiful addition! Thank you for suggesting it 🙌  , I did implement it and there are no differences in output `LGAT_snv_subset.tsv`

**Pull request merged**